src/Domain/ParallelDomain/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+set(LIBRARY ParallelDomain)
+
+add_spectre_library(${LIBRARY} INTERFACE)
+
+spectre_target_headers(
+  ${LIBRARY}
+  INCLUDE_DIRECTORY ${CMAKE_SOURCE_DIR}/src
+  HEADERS
+  DiagnosticInfo.hpp
+  FilterAction.hpp
+  FunctionsOfTimeAreReady.hpp
+  LimiterActions.hpp
```

Comment:
So these files don't really have a lot in common, and also don't necessarily have anything in common with Domain. However, each one depends on `Parallel` and was previously in a library that should *not* have depended on `Parallel`. So the "least bad" solution I could come up with was just put them all in their own library so that that library could depend on `Parallel`. However, I don't particularly like this solution.

Another option is to put each file into its own separate one-file library (in their respective directories) and just have those libraries depend on `Parallel`. Also not great.

Suggestions are welcome because I think this needs to be fixed.

Reply:
While not great, is there a problem depending on `Parallel`? It's very low level and shouldn't depend on `Domain`, so there's no cyclic dependency I think?

Reply:
Basically I interpret it as `Parallel` == Charm++ and the 4 libraries I moved these files from have nothing to do with charm at all (in my opinion) except for these actions.

Also, I think @nilsdeppe will have `Parallel` depend on `Domain` (or maybe `DomainStructure`?) for the nodegroup changes. I'm not sure on that.

Reply:
So here is my suggestions:
-  DiagnosticInfo:  should just be in Parallel.  Instead of passing in the Domain, just pass in the number of blocks
- the other three files I would move to `ParallelAlgorithms/Actions`

Reply:
I like that suggestion. Seems like the best option so far. Any objections @nilsvu @wthrowe @nilsdeppe ?

Reply:
No objections.

Reply:
Sounds fine to me!

Reply:
Ok pushed new commits to make these changes